### PR TITLE
[TextField] Fixed incorrect prefix spacing and coloring

### PR DESCRIPTION
**Pull request by MShwed**
### WHY are these changes introduced?

Resolves https://github.com/Shopify/polaris-react/issues/1132

### WHAT is this pull request doing?

- Adds spacing between TextField prefix and value
- String and Icon prefix values remain `color(ink, lighter)` regardless of TextField state.

Before:

Before-PR-Polaris

After:

After-PR-Polaris

## 

### How to 🎩

🖥 [Local development instructions](https://github.com/Shopify/polaris-react/blob/master/README.md#development)
🗒 [General tophatting guidelines](https://github.com/Shopify/polaris-react/blob/master/documentation/Tophatting.md)
📄 [Changelog guidelines](https://github.com/Shopify/polaris-react/blob/master/documentation/Versioning%20and%20changelog.md)

<details>
<summary>Copy-paste this code in <code>playground/Playground.tsx</code>:</summary>

```jsx
import * as React from 'react';
import {Page, Icon, FormLayout, TextField} from '../src';

interface State {
  prefixValue: string;
  iconPrefixValue: string;
}

export default class Playground extends React.Component<{}, State> {
  state = {
    number: '',
    iconNumber: '',
  };

  handleOnChange = (field: string) => (value: string) => {
    this.setState({[field]: value});
  };

  render() {
    return (
      <Page title="Playground">
        <FormLayout.Group condensed>
          <TextField
            type="text"
            onChange={this.handleOnChange('number')}
            label=""
            value={this.state.number}
            prefix="$"
          />
          <TextField
            type="text"
            onChange={this.handleOnChange('iconNumber')}
            label=""
            value={this.state.iconNumber}
            prefix={<Icon source="search" />}
          />
        </FormLayout.Group>
      </Page>
    );
  }
}
```

</details>

### 🎩 checklist

* [ ] Tested on [mobile](https://github.com/Shopify/polaris-react/blob/master/documentation/Tophatting.md#cross-browser-testing)
* [ ] Tested on [multiple browsers](https://help.shopify.com/en/manual/intro-to-shopify/shopify-admin/supported-browsers)
* [ ] Tested for [accessibility](https://github.com/Shopify/polaris-react/blob/master/documentation/Accessibility%20testing.md)

I’ve tophatted these changes in the following browsers:

- [x] Chrome latest
- [x] FF latest
- [x] Safari latest

### CODE REVIEW

File: src/components/TextField/TextField.scss [hunk 1]
```diff
@@ -2,6 +2,7 @@ $textfield-backdrop-offset: -(border-width(thick));
 $backdrop-horizontal-spacing: spacing(tight) * 1.5;
 $addon-horizontal-spacing: 0.25em;
 $spinner-icon-size: rem(12px);
+$prefix-horizontal-spacing: rem(8px);
```
Comment (dleroux) [inline src/components/TextField/TextField.scss]: Probably best to use `spacing(tight)` here.

File: src/components/TextField/TextField.scss [hunk 2]
```diff
@@ -165,9 +166,13 @@ $stacking-order: (
   position: relative;
   z-index: z-index(contents, $stacking-order);
   flex: 0 0 auto;
-  margin: 0 $addon-horizontal-spacing 0 $backdrop-horizontal-spacing;
-  color: currentColor;
+  margin: 0 $prefix-horizontal-spacing 0 $backdrop-horizontal-spacing;
+  color: color('ink', 'lighter');
```
Comment (dleroux) [inline src/components/TextField/TextField.scss]: I'm thinking we still would want the `color('ink', 'lightest`') for disabled fields @sarahill?

Comment (sarahill) [reply src/components/TextField/TextField.scss]: Good catch. Yes, we'll still want `color('ink', 'lightest')` for disabled fields.

**Comment by MShwed** [comment]
Allowed disabled TextFields to retain disabled style

Changes_Shopify_PR1

**Comment by danrosenthal** [comment]
@MShwed, are you ready for another round of review from @dleroux on this?

**Comment by MShwed** [comment]
@danrosenthal That would be great!

**Comment by dleroux** [comment]
Sorry about that. Reassigning me to the review when it's ready again will give me the necessary nudge for next time.

🎩 looks great! Thank you for your contribution @MShwed 🎉

**Comment by danrosenthal** [comment]
@MShwed we'd be happy to merge this change for you once you've resolved the conflict on `UNRELEASED.md`. I'd go ahead and do it for you if not for the complexity involved in updating a PR from a repo fork

**Comment by ry5n** [comment]
@MShwed Anything I can help with to get this over the finish line?

**Comment by danrosenthal** [comment]
@ry5n with no response from @MShwed, we should fork this and PR it ourselves

**Comment by danrosenthal** [comment]
Closing in favor of https://github.com/Shopify/polaris-react/pull/1597

**Pull request closed without merging**